src/main/java/com/connectfour/Player.java

```diff
@@ -3,19 +3,27 @@
 import java.util.Scanner;
 
 public class Player implements Playerable{
+    public final String VALIDATION_ERROR = "That is not a valid move! Please choose a number between 1 and 7";
+
     Discable disc;
     Scannable scanner;
     String name;
+    Validator moveValidator;
 
-    public Player(String name, Discable disc, Scannable scanner){
+    public Player(String name, Discable disc, Scannable scanner, Validator moveValidator){
         this.disc = disc;
         this.scanner = scanner;
         this.name = name;
+        this.moveValidator = moveValidator;
 
     }
 
-    public void takeTurn(Board board){
+    public void takeTurn(Displayable display, Board board){
         int colChoice = scanner.getColumn();
+        while(!moveValidator.isValid(board, colChoice)){
+            Presenter.printError(display, VALIDATION_ERROR);
+            colChoice = scanner.getColumn();
+        }
         colChoice = colChoice - 1;
         placeDisc(board, colChoice);
     }
```

Comment:
Interesting the player knows to reduce the colChoice by 1 and also places the disc.  Is this the responsibility of the player?  I would think the board would adjust the colChoice as needed - it's an implementation detail of the board.  And placing the disc is an orchestration of the game.

Reply:
Moved the -1 from Player to Board. Also removed the placeDisc function from Player, because I can just call Board#addDisc inside Player#takeTurn instead.

Reply:
board.addDisc really should not be the player's responsibility.  Should be the game's responsibility.  The player just provides it's move.  The game updates the board to reflect the player's move.

Reply:
I see what you mean. Fixed.